libbeat/processors/add_kubernetes_metadata/indexers_test.go

```diff
@@ -32,7 +32,7 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common/kubernetes"
 )
 
-var metagen = metadata.NewPodMetadataGenerator(common.NewConfig(), nil, nil, nil)
+var metagen = metadata.NewPodMetadataGenerator(common.NewConfig(), nil, nil, nil, nil)
```

Comment:
It can be confusing if `add_kubernetes_metadata` behaves differently, but I am ok with addressing this in a future change. For `add_kubernetes_metadata` we will need some cache for the replicasets to avoid requesting them on every event.

Reply:
We use the same wrapper function to initialise `PodMetadataGenerator` at https://github.com/elastic/beats/blob/20c7da2ce5beb52ef56dde7736eb4150deeaea57/libbeat/processors/add_kubernetes_metadata/kubernetes.go#L197 and because of this the `add_kubernetes_metadata` processor will have the same behaviour too. I explicitly tested that to verify it works like this and also that the metadata are cached and we hit k8s API only when on Pod's discovery.

Reply:
Thanks for the additional tests and clarifications, and sorry for the noise on this.